wgpu-hal/src/vulkan/device.rs

```diff
@@ -582,15 +582,26 @@ impl super::Device {
         let stage_flags = crate::auxil::map_naga_stage(naga_stage);
         let vk_module = match *stage.module {
             super::ShaderModule::Raw(raw) => raw,
-            super::ShaderModule::Intermediate(ref naga_shader) => {
+            super::ShaderModule::Intermediate {
+                ref naga_shader,
+                runtime_checks,
+            } => {
                 let pipeline_options = naga::back::spv::PipelineOptions {
                     entry_point: stage.entry_point.to_string(),
                     shader_stage: naga_stage,
                 };
+                let mut naga_options = self.naga_options.clone();
+                if !runtime_checks {
```

Comment:
Can we avoid cloning the options if this flag is true? Options contain a hashmap. This is not hugely important at this stage, but fixing seems easy still.